web/courses.go

```diff
@@ -533,6 +539,61 @@ func (s *AutograderService) setLastApprovedAssignment(submission *pb.Submission,
 	return s.db.UpdateEnrollment(query)
 }
 
+// acceptRepositoryInvites tries to accept repository invitations for the given repository types on behalf of the given user.
+func (s *AutograderService) acceptRepositoryInvites(ctx context.Context, sc scm.SCM, user *pb.User, course *pb.Course, repoTypes []pb.Repository_Type) {
+	invites := []*github.RepositoryInvitation{}
+	for _, repoType := range repoTypes {
+		repo, err := s.db.GetRepositoryByType(user.ID, course.OrganizationID, repoType)
+		if err != nil {
+			s.logger.Error("Failed to get repo: %s", err)
+			continue
+		}
+		// TODO: pb.Repository should contain Path
+		path := strings.Split(repo.GetHTMLURL(), "/")
+
+		repoInvites, err := sc.GetRepositoryInvites(ctx, &scm.RepositoryOptions{ID: repo.RepositoryID, Owner: course.OrganizationPath, Path: path[len(path)-1]})
```

Comment:
I'm a bit uncomfortable with the way you extract the path here. I'm not completely sure what it should be without testing things myself, but I see two solutions:
1. Add a `GetPath()` method in `ag/repo.go` to extract the path, you can add one that is similar to the `GetTestURL()`; this uses `LastIndex` instead of split, which is more efficient.
2. Another alternative is to only pass the RepoID and use the `GetRepository` method within `GetRepositoryInvites`. This is obviously less efficient, since it will talk with GitHub.

Reply:
I don't like it either. I was thinking that perhaps adding `Path` to the `pb.Repository` type and storing it in the database would be a good way to do this.

```go
// courses_new.go
// line 79:84
// repo scm.Repository
dbRepo := pb.Repository{
        OrganizationID: org.ID,
        RepositoryID:   repo.ID,
        HTMLURL:        repo.WebURL,
        RepoType:       pb.RepoType(path),
}
```
The `scm.Repository` already has Path, so we would just need to update the various functions that save repositories to the database.

Storing the `Path` is useful as we avoid having to extract anything from a string, and we already have to do a database lookup for the repository.

Then we could just do this:
```go
repoInvites, err := sc.GetRepositoryInvites(ctx, &scm.RepositoryOptions{ID: repo.RepositoryID, Owner: course.OrganizationPath, Path: repo.Path})
```

The use cases for this is obviously for this PR, in addition to, I believe, @oleespe's PR & Tasks work where having the path would be helpful.

This would not work for repositories already created (courses running this semester) without updating them in some way.

Alternatively I will go for your first suggested solution.

Reply:
Not sure we need to store it db, since it seems to contain the same info as HTMLURL minus some extra stuff... So I prefer to not add redundant data to the db. Note that, we may reconsider this later, e.g., after the semester, but it would require a more in-depth review of the existing usages of HTMLURL and Path etc. to see if some of these things can be harmonized and simplified. 

Note also that the doc comment for the `scm.RepositoryOptions` type; it says that you only need `ID` or both `Owner` and `Path`. That is, not all three. If you look at the other uses of this option, you'll see that id only `ID` is used we will first lookup the `Owner` and `Path` and use them to make the actual call to GitHub. (Exception: `GetRepository` can actually lookup the repository directly using only ID)

Suggest to go with my first suggestion for now.